File: GO/task1.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func GetRandomNumber(min, max int) int {
	rand.Seed(time.Now().UnixNano())
	return min + rand.Intn(max-min+1)
}

func task1() {
	fmt.Println("Task 1")
	quantityLine, quantityColumns := 5, 6
	maxSrednee, numberLine := 0, 0

	randomMatrix := make([][]int, quantityLine)
	for i := range randomMatrix {
		randomMatrix[i] = make([]int, quantityColumns)
		for j := range randomMatrix[i] {
			randomMatrix[i][j] = GetRandomNumber(0, 100)
		}
	}

	for _, i := range randomMatrix {
		for _, j := range i {
			fmt.Printf("%4d", j)
		}
		fmt.Println()
	}

	for i := 0; i < quantityLine; i++ {
		sredneeArithmetic := 0
		for j := 0; j < quantityColumns; j++ {
			sredneeArithmetic += randomMatrix[i][j]
		}
		if maxSrednee < sredneeArithmetic {
			maxSrednee = sredneeArithmetic
			numberLine = i
		}
		fmt.Printf("Номер строки %d среднее арифметическое = %.2f\n", i, float64(sredneeArithmetic)/float64(quantityColumns))
	}
	fmt.Printf("Наибольшее среднее арифметическое в строке %d\n", numberLine)
}

func task2() {
	fmt.Println("Task 2")

	sizeSquareMatrix := 15
	squareMatrix := make([][]int, sizeSquareMatrix)
	for i := range squareMatrix {
		squareMatrix[i] = make([]int, sizeSquareMatrix)
		for j := range squareMatrix[i] {
			squareMatrix[i][j] = GetRandomNumber(-100, 100)
			fmt.Printf("%4d", squareMatrix[i][j])
		}
		fmt.Println()
	}

	for i := 0; i < sizeSquareMatrix; i++ {
		columns := make([]int, sizeSquareMatrix)
		for j := 0; j < sizeSquareMatrix; j++ {
			columns[j] = squareMatrix[j][i]
		}
		if i%2 == 0 {
			Sort(columns, true)
		} else {
			Sort(columns, false)
		}
		for j := 0; j < sizeSquareMatrix; j++ {
			squareMatrix[j][i] = columns[j]
		}
	}

	fmt.Println()
	for _, i := range squareMatrix {
		for _, j := range i {
			fmt.Printf("%4d", j)
		}
		fmt.Println()
	}

	sumRight, sumLeft := 0, 0
	for i := 0; i < len(squareMatrix); i++ {
		for j := i; j < len(squareMatrix); j++ {
			if squareMatrix[i][j] > 0 {
				sumRight++
			}
		}
	}

	for i := len(squareMatrix) - 1; i >= 0; i-- {
		for j := i - 1; j >= 0; j-- {
			if squareMatrix[i][j] > 0 {
				sumLeft++
			}
		}
	}

	fmt.Printf("Количество положительных элементов правой половины = %d\tколичество положительных элементов левой половины = %d\n", sumRight, sumLeft)
	if sumRight > sumLeft {
		fmt.Println("Правая половина содержит больше положительных элементов")
	} else {
		fmt.Println("Левая половина содержит больше положительных элементов")
	}
}

func Sort(arr []int, asc bool) {
	if asc {
		for i := 0; i < len(arr)-1; i++ {
			for j := i + 1; j < len(arr); j++ {
				if arr[i] > arr[j] {
					arr[i], arr[j] = arr[j], arr[i]
				}
			}
		}
	} else {
		for i := 0; i < len(arr)-1; i++ {
			for j := i + 1; j < len(arr); j++ {
				if arr[i] < arr[j] {
					arr[i], arr[j] = arr[j], arr[i]
				}
			}
		}
	}
}

func main() {
	task1()
	fmt.Println()
	task2()
}

File: task1.cpp
#include <iostream>
#include <vector>
#include <string>
#include<random>
#include <algorithm>
#include <iomanip>
#include <Windows.h>

using namespace std;

int GetRandomNumber(int min, int max)
{
    random_device rd;//random_device, который является источником недетерминированных случайных чисел.
    //Затем мы используем это устройство для заполнения генератора std::minstd_rand. Затем функция генератора() используется для генерации случайного числа
    minstd_rand generator(rd());

    uniform_int_distribution<int> distribution(min, max);// функция destribition для задания диапозона значений
    int random_number = distribution(generator);
    return random_number;
}

void  task1()
{
    cout << "Task 1" << endl;
    int quantityLine = 5, quantityColumns = 6;
    int maxSrednee = 0, numberLine = 0;
    
    vector<vector<int>> randomMatrix(quantityLine, vector<int>(quantityColumns));;
  
    for (auto& i : randomMatrix)
    {
        for (auto& j : i)
        {
            j = GetRandomNumber(0, 100);
        }
    }
    for (auto& i : randomMatrix)
    {
        for (auto& j : i)
        {
            cout << setw(4) << j;
        }
        cout << endl;
    }

    for (int i = 0; i < quantityLine; i++)
    {
        float sredneeArithmetic = 0;

        for (int j = 0; j < quantityColumns; j++)
        {
            sredneeArithmetic += randomMatrix[i][j];// среднее арифетическое строки 
        }
        if (maxSrednee < sredneeArithmetic)// максимальное среднее арифметическое
        {
            maxSrednee = sredneeArithmetic;
            numberLine = i;
        }
        cout << "Номер строки " << i << " среднее арифметическое = " << sredneeArithmetic / quantityColumns << endl;
    }
    cout << "Наибольшее среднеее арифметическое в строке " << numberLine;
    cout << endl;
}


void task2()
{
    cout << "Task 2" << endl;

    int sizeSquareMatrix = 15;
    
    vector<vector<int>> squareMatrix(sizeSquareMatrix, vector<int>(sizeSquareMatrix));

    for (auto& i : squareMatrix)
    {
        for (auto& j : i)
        {
            j = GetRandomNumber(-100, 100);
            cout << setw(4) <<j;
        }
        cout << endl;
    }

    for (int i = 0; i < sizeSquareMatrix; i++)// цикл наоборот. Берем первый элемент строки
    {
        vector<int> columns(sizeSquareMatrix);// вектор для хранение стобцов
        for (int j = 0; j < sizeSquareMatrix; j++)// здесь рассматриваем строки
        {
           columns[j] = squareMatrix[j][i];// закидываем в вектор столбец все элементы первого столбца где j номер строки 
            
        }
        if (i % 2 == 0)// если строка четная то сортируем в порядке возрастания
        {
            sort(columns.begin(), columns.end());
        }
        else// если не четная то в порядке убывания
        {
            sort(columns.rbegin(), columns.rend());
        }
        for (int k = 0; k < sizeSquareMatrix; k++)
        {
            squareMatrix[k][i] = columns[k];
        }
    }
    cout << endl;

    for (auto& i : squareMatrix)
    {
        for (auto& j : i)
        {
            cout << setw(4) << j;
        }
        cout << endl;
    }

    int sumRigth = 0;

    for (int i = 0; i < squareMatrix.size(); i++)// высчитываем в правый угол относительно диагонали
    {
        for (int j = i; j < sizeSquareMatrix; j++)
        {
            if (squareMatrix[i][j] > 0) sumRigth++;
        }

    }
    
    int sumLeft = 0;

    for (int i = sizeSquareMatrix; i > 0; i--)// высчитываем в левый угол относительно диагонали
    {
        for (int j = i; j > 0; j--)
        {
            if (squareMatrix[i-1][j-1] > 0) sumLeft++;
        }

    }
    cout << "Количество положительных элементов правой половины = " << sumRigth << "\t количество положительных элементов левой половины = " << sumLeft << endl;
    if (sumRigth > sumLeft) cout << "Правая половина содержит больше положительных элементов "<<endl;
    else cout << "Левая половина содержит больше положительных элементов" << endl;
}

int main()
{
    SetConsoleCP(1251);
    SetConsoleOutputCP(1251);

    task1();// доделать какието элементы матрицы
    cout << endl;
    task2();


}